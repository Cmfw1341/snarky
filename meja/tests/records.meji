convertible type t('a, 'b, 'c) = {a: 'a, b: 'b, c: 'c}
with prover t('a, 'b, 'c) = {a: 'a, b: 'b, c: 'c}
by typ;

let x : t(int, int, int);

let y : t(bool, bool, bool);

let z : t(int, bool, unit);

module X :
  {
    convertible type t('a) = {a: 'a, b: 'a, c: 'a}
    with prover t('a) = {a: 'a, b: 'a, c: 'a}
    by typ;
    
    let x : t(int);
    
    };

let a : X.t(int);

let b : X.t(int);

let c : t(int, int, int);

let d : (int, int);

let e : int;

let f : bool;

let g : unit;

let h : unit;


