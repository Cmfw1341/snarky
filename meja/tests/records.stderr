File "tests/records.meja", line 5, characters 12-16:
Error: Unbound constructor true.
